src/AccessControl.jsx

```diff
@@ -45,11 +56,44 @@ function AccessControl() {
     }
   };
 
+  const approvedWallets = useQuery({
+    queryKey: ['approved-wallets'],
```

Comment:
For all queries please make sure to include chainId in cache key

Reply:
added chainId to all queries